lib/ansible/modules/database/postgresql/postgresql_pg_hba.py

```diff
@@ -0,0 +1,673 @@
+        else:
+            sourceobj = self.source()
+            if isinstance(sourceobj, ipaddress.IPv4Network):
+                srcweight = sourceobj.prefixlen * 4
+            elif isinstance(sourceobj, ipaddress.IPv6Network):
+                srcweight = sourceobj.prefixlen
+            elif isinstance(sourceobj, str):
+                # You can also write all to match any IP address,
+                # samehost to match any of the server's own IP addresses,
+                # or samenet to match any address in any subnet that the server is connected to.
+                if sourceobj == 'all':
+                    # (all is considered the full range of all ips, which has a weight of 0)
+                    srcweight = 0
+                elif sourceobj == 'samehost':
+                    # (sort samehost second after local)
+                    srcweight = 129
+                elif sourceobj == 'samenet':
+                    # Might write some fancy code to determine all prefix's
+                    # from all interfaces and find a sane value for this one.
+                    # For now, let's assume IPv4/24 or IPv6/96 (both have weight 96).
+                    srcweight = 96
+                elif sourceobj[0] == '.':
+                    # suffix matching (domain name), let's asume a very large scale
+                    # and therefore a very low weight IPv4/16 or IPv6/64 (both have weight 64).
+                    srcweight = 64
+                else:
+                    # hostname, let's asume only one host matches, which is
+                    # IPv4/32 or IPv6/128 (both have weight 128)
+                    srcweight = 128
+
+        if self['db'] == 'all':
+            dbweight = numdbs
+        elif self['db'] == 'replication':
+            dbweight = 0
+        elif self['db'] in ['samerole', 'samegroup']:
+            dbweight = 1
+        else:
+            dbweight = 1 + self['db'].count(',')
+
+        if self['usr'] == 'all':
+            uweight = numusers
+        else:
+            uweight = 1
+
+        ret = []
+        for character in order:
+            if character == 'u':
+                ret.append(uweight)
+            elif character == 's':
+                ret.append(srcweight)
+            elif character == 'd':
+                ret.append(dbweight)
+        ret.append(sourceobj)
+
+        return tuple(ret)
+
+
+def main():
+    '''
+    This function is the main function of this module
+    '''
+    module = AnsibleModule(
+        argument_spec=dict(
+            address=dict(type='str', default='samehost', aliases=['source', 'src']),
+            contype=dict(type='str', default=None, choices=PG_HBA_TYPES + ['']),
+            create=dict(type='bool', default=False),
+            databases=dict(type='str', default='all'),
+            dest=dict(type='path', required=True),
+            method=dict(type='str', default='md5', choices=PG_HBA_METHODS),
+            netmask=dict(type='str', default=''),
```

Comment:
```suggestion
            netmask=dict(type='str'),
```
seems it's default='' by default